packages/workflow/src/interceptors.ts

```diff
@@ -137,19 +138,55 @@ export interface WorkflowOutboundCallsInterceptor {
   ) => Promise<[Promise<string>, Promise<unknown>]>;
 }
 
+/** Input for WorkflowInternalsInterceptor.concludeActivation */
+export interface ConcludeActivationInput {
+  commands: coresdk.workflow_commands.IWorkflowCommand[];
+}
+
+/** Output for WorkflowInternalsInterceptor.concludeActivation */
+export type ConcludeActivationOutput = ConcludeActivationInput;
+
+/** Input for WorkflowInternalsInterceptor.activate */
+export interface ActivateInput {
+  activation: coresdk.workflow_activation.IWFActivation;
+  jobIndex: number;
+}
+
+/**
+ * Interceptor for the internals of the Workflow runtime.
+ *
+ * Use to manipulate or trace Workflow activations.
+ */
+export interface WorkflowInternalsInterceptor {
+  /**
+   * Called when the Workflow runtime runs a WFActivationJob.
+   *
+   * Returns a `boolean` indicating whether this job was processed or not
+   */
+  activate?(input: ActivateInput, next: Next<this, 'activate'>): Promise<boolean>;
+  /**
+   * Called after all `WFActivationJob`s have been proccesed for an activation.
+   *
+   * Can manipulate the commands generated by the Workflow
+   */
+  concludeActivation?(input: ConcludeActivationInput, next: Next<this, 'concludeActivation'>): ConcludeActivationOutput;
```

Comment:
Should these be "activated" and "concluded"?

Reply:
These are the names of the methods in the Workflow runtime, I'd rather keep them.